src/Arcus.Messaging.Abstractions/MessageHandling/MessageHandler.cs

```diff
@@ -129,17 +139,40 @@ private static IEnumerable<Type> GetServiceRegistrationTypesFromMicrosoftExtensi
             return serviceTypes;
         }
 
+        private static object GetEngineFromServiceProviderNet6(IServiceProvider serviceProvider)
+        {
+            object engine = serviceProvider.GetOptionalPropertyValue("RootProvider", BindingFlags.NonPublic | BindingFlags.Instance)?.GetOptionalFieldValue("_engine");
+            return engine;
+        }
+
+        private static object GetCallSiteFactoryFromServiceProviderNetCore(object engine)
+        {
+            object callSiteFactory = engine.GetOptionalPropertyValue("CallSiteFactory", BindingFlags.NonPublic | BindingFlags.Instance);
+            return callSiteFactory;
+        }
+
+        private static object GetCallSiteFactoryFromServiceProviderNet6(object engine)
+        {
+            object callSiteFactory =
+                engine.GetOptionalFieldValue("_serviceProvider")
+                      .GetOptionalPropertyValue("CallSiteFactory", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            return callSiteFactory;
+        }
+
         private static IEnumerable<Type> GetServiceRegistrationTypesFromWebJobs(IServiceProvider serviceProvider, ILogger logger)
         {
-            object container = serviceProvider.GetType().GetField("_container", BindingFlags.NonPublic | BindingFlags.Instance)?.GetValue(serviceProvider);
+            object containerNetCore = serviceProvider.GetOptionalFieldValue("_container");
+            object containerNet6 = serviceProvider.GetOptionalFieldValue("_resolver");
+            object container = containerNetCore ?? containerNet6;
```

Comment:
For me, this is also a bit obscure.  Why the fields _resolver and _container ?  And how are they related with 'containerNetCore' and containerNet6 ?

Reply:
This is because the fields were renamed and that versions. So, in .NET Core it was called `_container` but in .NET 6 they renamed the field to `_resolver`.

Reply:
It's not very stable, I know, it's the best we can do though. We can/should discus this later, for sure.